### ISSUE TITLE
WezTerm should use Xcursor theme on Wayland

### ISSUE THREAD

**Issue by jgcodes2020** [wez/wezterm#1742]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### WezTerm version

wezterm 20220319-163727-ee5d5714

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

#524 reported the same bug on X11. Although Wayland does not have a cursor configuration protocol just yet ([see here](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/21) for some work being done), the de-facto configuration method is still to use the Xcursor theme.

Setting the relevant environment variables `XCURSOR_THEME` and `XCURSOR_SIZE` does change the cursor, so there's that.

### To Reproduce

- Use a different cursor theme. (Ubuntu uses one by default)
- Move the cursor over the WezTerm window.
- Notice that the cursor is now the default one. (For me this is Breeze because I installed some KDE software)

### Configuration

The only relevant setting is `enable_wayland = true`.

### Expected Behavior

This is my normal cursor (Yaru):
![image](https://user-images.githubusercontent.com/35639051/159196694-295c2283-ea6e-432e-a189-29b5c631fb41.png)

This is the cursor I see when mousing over WezTerm (Breeze):
![image](https://user-images.githubusercontent.com/35639051/159196745-035725f7-d5a7-4ea6-8eb0-62fdbce06945.png)

I would expect to not see the cursor change when mousing over WezTerm

### Logs

Logs don't provide relevant info

**Comment by wez** [comment]
`main` is now technically capable of talking to [XDG Desktop Portal](https://flatpak.github.io/xdg-desktop-portal/) to query these sorts of settings. The settings interface currently only portably defines the dark mode/appearance in a DE-agnostic way; these are the portable settings:

```
gdbus call --session --dest=org.freedesktop.portal.Desktop --object-path=/org/freedesktop/portal/desktop --method=org.freedesktop.portal.Settings.ReadAll '["org.freedesktop.appearance"]'
({'org.freedesktop.appearance': {'color-scheme': <uint32 1>}, 'org.freedesktop.appearance': {'color-scheme': <uint32 1>}, 'org.freedesktop.appearance': {'color-scheme': <uint32 1>}},)
```

However, the cursor theme and size are reported under a `org.gnome.desktop.interface` namespace along with a lot of other gnome specific stuff; you can see for yourself by running:

```
gdbus call --session --dest=org.freedesktop.portal.Desktop --object-path=/org/freedesktop/portal/desktop --method=org.freedesktop.portal.Settings.ReadAll '[]'
```

It wouldn't be terribly difficult to hook this up in wezterm, but I'm not sure if this would give surprising results sometimes. For example, if you have two sessions running different DE on the same machine, would they both end up using the gnome settings? That's probably not desirable. I don't know if this stuff is smart enough to avoid that or not.

**Comment by MuhammedZakir** [comment]
> For example, if you have two sessions running different DE on the same machine, would they both end up using the gnome settings?

I don't think that will cause problem because if I understand correctly, when you query the session dbus (`--session`), it gets the session address from `DBUS_SESSION_BUS_ADDRESS` env var, and different login sessions will have different session addresses.

**Comment by miguno** [comment]
The code snippet at https://github.com/wez/wezterm/issues/1742#issuecomment-1075333507, which explicitly sets wezterm's `xcursor_theme` and `xcursor_size` configs based on GNOME's existing settings for those configs, is the only solution I found to fix the mouse cursor from disappearing as described at https://github.com/wez/wezterm/issues/3334.

I would have added this comment to the linked issue, but couldn't because that issue is closed. Maybe @wez could add that, as the linked issue is the first Google result when searching for e.g. "wezterm mouse disappearing"?

**Comment by wez** [comment]
Done

**Comment by GottZ** [comment]
> It wouldn't be terribly difficult to hook this up in wezterm, but I'm not sure if this would give surprising results sometimes. For example, if you have two sessions running different DE on the same machine, would they both end up using the gnome settings? That's probably not desirable. I don't know if this stuff is smart enough to avoid that or not.

for some reason it feels like the sane default to please the majority of users by attempting to query gsettings (which also works on kde btw), thus showing the wrong cursor to a minority of users, outweighs showing the wrong cursor to almost all users by default.

if someone is really running multiple dm's at once, they are (in my opinion) more likely to help figuring this out than the classic ubuntu user.

**Comment by dhouck** [comment]
[The recommended config change](<https://github.com/wezterm/wezterm/issues/1742#issuecomment-1075333507>) above worked, even though I use KDE and donʼt think Iʼve ever used GNOME Wayland on this computer.  Iʼm not sure why it isnʼt the default; every other app Iʼve tried uses the right cursors.

However,
> In addition to no longer requiring a wrapper script, this has the advantage of you being able to manually trigger a config reload (`ctrl-shift-r` by default) to update the theme without having to restart wezterm.

This didnʼt work for me; I needed to actually restart Wezterm to get it to work.  Every other app I use, again, uses the new settings instantly without a restart or a manual configuration reload.

I donʼt know much about Wayland protocols but it looks like almost every compositor implements [`cursor_shape_v1`](https://wayland.app/protocols/wayland-protocols/302) and I see articles saying itʼs what other programs use.

EDIT TO ADD: I found [this article](https://blogs.kde.org/2024/10/09/cursor-size-problems-in-wayland-explained/) which mentions both the protocol mentioned above (which wasnʼt as widely supported at the time as it is now), and the other standard options.

(Now if only I could find a cursor theme I particularly liked instead of a bunch clearly designed by people with ~~bad design sense~~ different aesthetics than I have, that would be nice, but I donʼt think WezTerm is going to help much there)

**Comment by romain145** [comment]
The recommended config change doesn't work on Linux Mint Cinnamon for a high-DPI display that is scaled 200%.

I was able to force the cursor size using this config:

```lua
-- Flatpak sandbox can't read host gsettings, so set these explicitly.
-- Theme must live in ~/.icons (sandbox has home:ro, not /usr/share/icons).
return {
  xcursor_theme = "Bibata-Modern-Classic",
  xcursor_size = 48,
}
```
